src/main/java/org/example/bookshop/service/impl/BookServiceImpl.java

```diff
@@ -26,6 +27,9 @@ public class BookServiceImpl implements BookService {
 
     @Override
     public BookDto save(CreateBookRequestDto book) {
+        if (bookRepository.existsByIsbn(book.getIsbn())) {
```

Comment:
Initially, I wanted to check the isbn and email using my own validator, but when I asked QnA about a registration exception that needed to be thrown when we receive the same email, the mentor said that we shouldn't use such complex methods as a database query in the validation annotation.

Reply:
So what finally should I do?

Reply:
Actually, you can't save the same isbn in DB, because you should have constraint Column(unique = true), so this checking is redundant and reduce performance. So you can just remove this checking everywhere